notification/src/main/java/org/killbill/billing/notification/plugin/api/TagMetadata.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 Groupon, Inc
+ * Copyright 2022 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.notification.plugin.api;
+
+import java.util.UUID;
+
+import org.killbill.billing.ObjectType;
+import org.killbill.billing.util.tag.TagDefinition;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class TagMetadata {
+
+	private final UUID tagId;
+	private final UUID objectId;
+	private final ObjectType objectType;
+	private final TagDefinition tagDefinition;
+
```

Comment:
The `objectId` and `objectType`  would anyway be present in the event object being created, so I think they need not be present in this class. Also, is the `tagId` required? The [original code](https://github.com/killbill/killbill/blob/4da84cfb3722575d7c232fbae1d4e01d0dbfbef4/beatrix/src/main/java/org/killbill/billing/beatrix/extbus/BeatrixListener.java#L319) seems to be only returning the `tagDefinitionName` as a String.

Reply:
Hi @reshmabidikar 

objectType in the event object and that being saved in metadata are different. Please check the value saved for ACCOUNT type TAG in DB

`{"objectId":"df552482-ad3b-4feb-8fe2-d89fbc92316d",
**"objectType":"TAG",**
"eventType":"TAG_DELETION",
"accountId":"3f7bc813-fdaf-439d-bd80-bcce33cc856c",
"tenantId":"81fbdd60-9d0c-43ec-b05d-9be9920c627c",
"metaData":"{\"tagId\":\"df552482-ad3b-4feb-8fe2-d89fbc92316d\",\"objectId\":\"3f7bc813-fdaf-439d-bd80-bcce33cc856c\"**,\"objectType\":\"ACCOUNT\",**\"tagDefinition\":{\"@class\":\"org.killbill.billing.util.tag.DefaultTagDefinition\",\"id\":\"538de529-bcbc-4532-b644-025ebe261968\",\"name\":\"stringg test\",\"description\":\"stringg test\",\"controlTag\":false,\"applicableObjectTypes\":[\"ACCOUNT\"],\"createdDate\":null,\"updatedDate\":null}}"}`

Reply:
Since the [original code](https://github.com/killbill/killbill/blob/4da84cfb3722575d7c232fbae1d4e01d0dbfbef4/beatrix/src/main/java/org/killbill/billing/beatrix/extbus/BeatrixListener.java#L319)  only returns the `tagDefinitionName` as a String, would it not be sufficient to just include this field in the `TagMetadata` object? So, are `objectId`, `objectType` and `tagId` required?

Reply:
Hi @reshmabidikar 

Updated the code as per comments